src/nemos/solvers.py

```diff
@@ -0,0 +1,725 @@
+from functools import partial
+from typing import Any, Callable, NamedTuple, Optional, Union
+
+import jax
+import jax.flatten_util
+import jax.numpy as jnp
+from jax import grad, jit, lax, random
+from jaxopt import OptStep
+from jaxopt._src import loop
+from jaxopt.prox import prox_none
+
+from .tree_utils import tree_add_scalar_mul, tree_l2_norm, tree_slice, tree_sub
+from .typing import KeyArrayLike, Pytree
+
+
+class SVRGState(NamedTuple):
+    iter_num: int
+    key: KeyArrayLike
+    error: float
+    stepsize: float
+    xs: Optional[Pytree] = None
+    df_xs: Optional[Pytree] = None
```

Comment:
I think this naming is unclear. is it documented anywhere? can we give this class a docstring explaining what they are? and is this naming matching a convention? if so, should point that out. if not, I think it should be changed to something more clear

Reply:
They are documented in the implementation of the algorithm, but we can add a docstring here as well.
The naming is trying to match the pseudocode provided in [this paper](https://arxiv.org/abs/2010.00892).